crates/matrix-sdk-ffi/src/api.udl

```diff
@@ -1,13 +1,19 @@
 namespace sdk {
     [Throws=ClientError]
-    Client login_new_client(string base_path, string username, string password);
+    Client new_client(string homeserver);
+
+    [Throws=ClientError]
+    Client discover_client(string domain);
```

Comment:
Ideally there would just be a single function here, that attempted to discover the server and if that failed, checked the domain to see if it is in fact the server. so the user could enter either `example.com` or `matrix.example.com` and the SDK would do the right thing. (I've handled this in EXI for now though as I wasn't sure how to achieve that).